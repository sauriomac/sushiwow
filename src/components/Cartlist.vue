<template>
  <transition name="fade">
    <div id="principal" class="principal mt-5">
      <div class="titulo">
        <h1 class="text-warning">{{ title }}</h1>
      </div>
      <div class="contenedor">
        <div
          class="col-md-3 col-xs card p-0 shadow"
          v-for="(item, id) in productos"
          v-bind:key="id"
        >
          <img class="card-img-top" v-bind:src="item.imgURL" alt="" />
          <div class="card-body">
            <router-link to="/ProductDetail">
              <h5
                class="card-title text-warning font-weight-bold"
                @click="addtodetail(item)"
              >
                {{ item.nombre }}
              </h5>
            </router-link>
            <small class="card-text text-white">
              {{ item.description }}
            </small>
            <h5 class="card-text text-warning font-weight-bold">
              {{ item.precio | currency }}
            </h5>
            <button
              class="btn bg-white text-red font-weight-bold"
              @click="addtocart(item)"
            >
              В КОРЗИНУ +
            </button>
            
          </div>
        </div>
      </div>
    </div>
  </transition>
</template>

<script>
import { mapMutations, mapActions } from "vuex";
export default {
  props: ["productos", "title"],
  name: "CartList",

  computed: {},

  methods: {
    ...mapMutations(["ADD_TO_CART"]),
    ...mapActions(["AddProduct"]),

    addtocart(item) {
      this.ADD_TO_CART(item);
    },
    addtodetail(item) {
      this.AddProduct(item);
    },
  },
};
</script>
<style>
.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.35s ease-out;
}

.fade-enter,
.fade-leave-to {
  opacity: 0;
}
</style>
