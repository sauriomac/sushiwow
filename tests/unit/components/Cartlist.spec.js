import { shallowMount, mount } from '@vue/test-utils'
import Cartlist from '@/components/Cartlist'


describe('Cartlist',()=>{

    test('would make match with snapshot',() => {

        const wrapper = shallowMount( Cartlist)

        expect( wrapper.html() ).toMatchSnapshot()

    })




})