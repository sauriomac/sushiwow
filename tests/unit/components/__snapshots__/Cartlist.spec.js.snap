// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cartlist would make match with snapshot 1`] = `
<transition-stub name="fade">
  <div id="principal" class="principal mt-5">
    <div class="titulo">
      <h1 class="text-warning"></h1>
    </div>
    <div class="contenedor"></div>
  </div>
</transition-stub>
`;
